README.md

```diff
@@ -1,22 +1,63 @@
 # tf_cod
 Terraform repo used for Clusters On Demand (COD)
 
-## Branches
+## Smoketests
 
-<table>
-<tr>
-<th>Branch</th>
-<th>Description</th>
-</tr>
-<tr>
-<td>main</td>
-<td>Used by `dev-ci` cluster</td>
-</tr>
+Whenever `tf_cod` has a commit to it, a Dockerfile located in [smoketests/Dockerfile](smoketests/Dockefile) gets built via a [drone job](https://drone.devops.indico.io/IndicoDataSolutions/tf_cod)
 
-<tr>
-<td>1.0.15</td>
-<td>Used by 5.7.0 cluster</td>
-</tr>
+This Dockerfile contains the smoketests that validate that the input variables [./variables.tf](variables.tf) are indeed used to configure the infrastructure.  The Dockerfile contains the `aws` cli as well as the `az` cli, along with `kubectl`.  The tests may freely use these cli tools since there is a policy which enables it to run without special authentication in the same way that the boto3 blob api works.
 
+The (pytest) tests are in 3 categories:
 
-</table> 
+1. AWS Tests located in [smoketests/aws/test_aws.py](smoketests/aws/test_aws.py)
+2. Azure Tests located in [smoketests/azure/test_azure.py](smoketests/azure/test_azure.py)
+3. Common Tests located in [smoketests/common/test_common.py](smoketests/common/test_common.py)
+
+### Automatic Variable Mapping
+
+Whenever the `tf_cod` has a commit, a `pre-commit` hook runs and generates the file called [tf-smoketest-variables.tf](./tf-smoketest-variables.tf) which will contain a configmap mapping all variables to their supplied values, for example:
```

Comment:
Is there some documentation that explains how to ensure that pre-commit functionality is enabled on a local developer environment?